devtools/common/src/main/java/io/quarkus/generators/SourceType.java

```diff
@@ -68,4 +73,8 @@ private String computeTemplateFile(String genericTemplate, String templateName)
     private String computeTemplateFile(String genericTemplate, String templateName, String fileName) {
         return String.format(genericTemplate, templateName, getPathDiscriminator(), fileName);
     }
+
+    public static Optional<SourceType> parse(String extension) {
+        return Arrays.stream(values()).filter(e -> e.extension.contains(extension)).findAny();
```

Comment:
This won't work. The extension we're talking about is the file extension so in the case of Kotlin, it's `.kt` so it doesn't contain `kotlin`.

I think I would have a method in the enum matching the extension list itself so that we can take a decision based on the list of extensions (it won't be useful for now but we could imagine it could).

Reply:
I have fixed it by comparing it with name.